src/neuroglancer/webgl/offscreen.ts

```diff
@@ -217,35 +217,34 @@ export class FramebufferConfiguration<ColorBuffer extends TextureBuffer|Renderbu
    * Only supports UNSIGNED_BYTE RGBA textures.
    */
   readPixel(textureIndex: number, glWindowX: number, glWindowY: number): Uint8Array {
-    let {gl} = this;
-    try {
-      this.bindSingle(textureIndex);
-      gl.readPixels(glWindowX, glWindowY, 1, 1, gl.RGBA, gl.UNSIGNED_BYTE, tempPixel);
-    } finally {
-      this.framebuffer.unbind();
-    }
-    return tempPixel;
+    return this.readPixels(textureIndex, glWindowX, glWindowY, 1, 1, tempPixel);
   }
 
   readPixels(
     textureIndex: number, glWindowX: number, glWindowY: number, 
-    width: number, height: number
+    width: number, height: number, buffer?: Uint8Array
   ) : Uint8Array {
 
     let {gl} = this;
-    let cutout = new Uint8Array(width * height * 4);
-
-    let left = Math.max(0, glWindowX - (width >> 1))
-    let bottom = Math.max(0, glWindowY - (height >> 1))
+    
+    if (!buffer) {
+      buffer = new Uint8Array(width * height * 4);
+    }
+    
+    // Appearently WebGL supports reading pixels off the 
```

Comment:
Supports, but doesn't necessarily zero them out. From the [specification](https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glReadPixels.xml):
> Values for pixels that lie outside the window connected to the current GL context are undefined.

Reply:
Thanks for the link, I couldn't find that spec before.

Reply:
Hmm, this problem is probably true in the positive direction as well.